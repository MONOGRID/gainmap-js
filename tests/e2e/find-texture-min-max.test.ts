import 'jest-extended'

import { describe, expect, it } from '@jest/globals'
import { toMatchImageSnapshot } from 'jest-image-snapshot'

import { getPage } from './common'

expect.extend({ toMatchImageSnapshot })

const matrix = [
  ['memorial.exr'],
  ['memorial.hdr'],
  ['chcaus2-bloom.exr'],
  ['chcaus2-bloom.hdr'],
  ['spruit_sunrise_1k.hdr']
]

describe('find-texture-min-max', () => {
  it.each(matrix)('finds min and max in %p', async (file) => {
    // we need to launch puppeteer with a
    // custom written "testbed.html" page
    // because our lib works by
    // rendering the SDR image with THREEjs
    // which only works in webgl (not here in node where we test)
    const { page, pageError, pageLog } = await getPage('find-texture-min-max')

    // we receive Arrays because puppeteer can't transfer Uint8Array data
    const result = await page.evaluate(`findTextureMinMax('${file}')`) as { min: number, max: number }

    expect(pageError).not.toBeCalled()
    // expect no calls to page log except the one indicated
    expect(pageLog).not.toBeCalledWith(expect.not.stringMatching(/GPU stall due to ReadPixels/))

    expect(result).toMatchSnapshot()
  }, 900000 /* 15 minutes */)
})
