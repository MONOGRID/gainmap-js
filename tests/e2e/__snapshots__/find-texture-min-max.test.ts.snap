// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find-texture-min-max finds min and max in "chcaus2-bloom.exr": min & max 1`] = `
{
  "max": [
    94.875,
    94.875,
    94.875,
  ],
  "min": [
    0,
    0,
    0,
  ],
}
`;

exports[`find-texture-min-max finds min and max in "chcaus2-bloom.hdr": min & max 1`] = `
{
  "max": [
    94.8125,
    94.8125,
    94.8125,
  ],
  "min": [
    0,
    0,
    0,
  ],
}
`;

exports[`find-texture-min-max finds min and max in "memorial.exr": min & max 1`] = `
{
  "max": [
    274,
    218,
    126,
  ],
  "min": [
    0,
    0,
    0,
  ],
}
`;

exports[`find-texture-min-max finds min and max in "memorial.hdr": min & max 1`] = `
{
  "max": [
    275,
    218.75,
    126.4375,
  ],
  "min": [
    0,
    0,
    0,
  ],
}
`;

exports[`find-texture-min-max finds min and max in "spruit_sunrise_1k.hdr": min & max 1`] = `
{
  "max": [
    65504,
    65504,
    44192,
  ],
  "min": [
    0,
    0,
    0,
  ],
}
`;
